mesonbuild/dependencies/mpi.py

```diff
@@ -37,65 +38,78 @@ def mpi_factory(env: 'Environment',
         return []
     compiler_is_intel = compiler.get_id() in {'intel', 'intel-cl'}
 
-    # Only OpenMPI has pkg-config, and it doesn't work with the intel compilers
-    if DependencyMethods.PKGCONFIG in methods and not compiler_is_intel:
-        pkg_name = None
+    if DependencyMethods.CONFIG_TOOL in methods:
+        nwargs = kwargs.copy()
```

Comment:
The commit message contains extremely little content at all, and doesn't mention that this is changing the order to look for the `mpi${lang} -showme` configtools before pkg-config. Why?

Reply:
I was in a hurry so I quickly committed. This PR is a draft and is not ready at all. I created it only to mention that I'm working on this subject. I'm going to tell you when it is in a better state.

Reply:
Ah ok, thanks for clarifying.

Reply:
I think this is now ready to be reviewed.

Reply:
I'd suggest, for reviewability, to change the order in one commit, and "support more implementations" in a second commit -- as they seem to be logically distinct changes which you are trying to make.

(I'd also like to hear more about *why* you are changing the order. :))

Reply:
I split the first commit. I try to explain why mpicc has to be considered first (before pkg-config) in a commit message.